filter_not_in_servicetype_list.go

```diff
@@ -0,0 +1,47 @@
+package aiven
+
+var _ Filter = (*NotInServiceTypeListFilter)(nil)
+
+type (
+	// List is a list of strings.
+	List map[string]struct{}
+
+	// NotInServiceTypeListFilter is a filter implementation that filters out elements that are not in the given list.
+	NotInServiceTypeListFilter struct {
+		BasicFilter
+
+		// list is a List of elements that are not filtered out.
+		list List
+	}
+)
+
+// Filter filters the given elements.
+func (f *NotInServiceTypeListFilter) Filter(fi Filterable) Filter {
+	if f.copy(fi); f.result.err != nil {
+		return f.setResult(nil, f.result.err)
+	}
+
+	afi, ok := f.filterableCopy.(map[string]ServiceType)
+	if !ok {
+		return f.setResult(nil, errTypeNotSupported)
+	}
+
+	for k := range afi {
+		if _, ok = f.list[k]; !ok {
```

Comment:
here's the reason we use maps instead of slices, this is going to work faster, and we also avoid additional map to slice conversion as we get the original data in the form of `map[string]ServiceType` from the endpoint

Reply:
this is no longer relevant